File: SingleStepperSlider/SingleStepperSliderAMIB1/states.cpp
#include "states.h"

static const StateInfo state_infos[2] = {
  {IDLE::setup, IDLE::enter, IDLE::exit, IDLE::loop, IDLE::event},
  {SINGLESTEPPERSLIDER::setup, SINGLESTEPPERSLIDER::enter, SINGLESTEPPERSLIDER::exit, SINGLESTEPPERSLIDER::loop, SINGLESTEPPERSLIDER::event}
};

static const WireValue wire_values[3] = {
  {1, 0, sizeof(uint32_t), (Value<void*>*) &SINGLESTEPPERSLIDER::position},
  {1, 1, sizeof(uint32_t), (Value<void*>*) &SINGLESTEPPERSLIDER::speed},
  {1, 2, sizeof(uint32_t), (Value<void*>*) &SINGLESTEPPERSLIDER::acceleration}
};

MasterManager<State, 2, 3> manager(0x6a19ede2, state_infos, wire_values, 0);

namespace IDLE {


void event(uint8_t ev) {
  switch (ev) {
  
  default:
    break;
  }
}


}
namespace SINGLESTEPPERSLIDER {
Value<uint32_t> position;
Value<uint32_t> speed;
Value<uint32_t> acceleration;

void event(uint8_t ev) {
  switch (ev) {
  case 0:
    events::homeStepper();
    break;
  case 1:
    events::positiveMove();
    break;
  case 2:
    events::negativeMove();
    break;
  case 3:
    events::enableStepper();
    break;
  case 4:
    events::disableStepper();
    break;
  default:
    break;
  }
}


}

File: SingleStepperSlider/states.js

var IDLE = {
  id: 0,
  master: {
    values: {
      
    },
    events: {
      
    }
  },
  tablet: {
    values: {
      
    },
    events: {
      
    }
  }
};
var STATE_IDLE = 0;
var SINGLESTEPPERSLIDER = {
  id: 1,
  master: {
    values: {
      position: new HardwareValue(1, 0, Manager.TYPE_UINT32),
      speed: new HardwareValue(1, 1, Manager.TYPE_UINT32),
      acceleration: new HardwareValue(1, 2, Manager.TYPE_UINT32)
    },
    events: {
      homeStepper: function homeStepper() { manager.sendEvent(0, 1); },
      positiveMove: function positiveMove() { manager.sendEvent(1, 1); },
      negativeMove: function negativeMove() { manager.sendEvent(2, 1); },
      enableStepper: function enableStepper() { manager.sendEvent(3, 1); },
      disableStepper: function disableStepper() { manager.sendEvent(4, 1); }
    }
  },
  tablet: {
    values: {
      
    },
    events: {
      
    }
  }
};
var STATE_SINGLESTEPPERSLIDER = 1;

var STATES = {
  IDLE: IDLE,
  SINGLESTEPPERSLIDER: SINGLESTEPPERSLIDER
};
var manager = new Manager([IDLE, SINGLESTEPPERSLIDER]);
